### PR TITLE
add nonuser screen for mobile search

### PR DESCRIPTION
**Pull request by chrisnojima**
We didn't have the not-a-keybase-user profile screen on mobile.
In desktop we have a user pane on the right side that's _kinda_ like the profile page but not really. We likely should have just controlled that with a flag. Anyways, on desktop we load this pane but on mobile we push the actual profile page. There i no non-user profile page (there will be one in search v3) so short term i ported the non-user search side page and in the mobile context it can push that as as a route. This is really a short term thing before search v3.

@keybase/react-hackers

### CODE REVIEW

File: shared/search/container.js [hunk 1]
```diff
@@ -36,6 +36,22 @@ const mapStateToProps = (state: TypedState) => {
   }
 }
 
+const mobileOnClickUserInGroup = (dispatch, user) => {
```
Comment (chrisnojima) [inline shared/search/container.js]: more complicated so pulled it up. If its a normal profile do the old stuff. if its an external one push a nonProfile

File: shared/search/user-pane/non-user.container.js [hunk 2]
```diff
@@ -0,0 +1,26 @@
+// @flow
+import NonUser from './non-user.render'
+import {connect} from 'react-redux'
+
+import type {TypedState} from '../../constants/reducer'
+
+const mapStateToProps = (state: TypedState, {routeProps}) => ({
```
Comment (chrisnojima) [inline shared/search/user-pane/non-user.container.js]: little wrapper to connect this thing

File: shared/search/user-pane/non-user.render.native.js [hunk 3]
```diff
@@ -0,0 +1,75 @@
+// @flow
+import React from 'react'
```
Comment (chrisnojima) [inline shared/search/user-pane/non-user.render.native.js]: mostly a copy / paste of non-user.render.desktop

**Comment by chrisnojima** [comment]
@keybase/react-hackers bump

File: shared/search/user-pane/non-user.container.js [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+  serviceName: routeProps.serviceName,
+  username: routeProps.username,
+  title: routeProps.username,
+})
+
+const mapDispatchToProps = (dispatch: Dispatch, {navigateUp}) => ({
+  onBack: () => dispatch(navigateUp()),
+})
+
+const mergeProps = (stateProps, dispatchProps, ownProps): any => ({
```
Comment (gabriel) [inline shared/search/user-pane/non-user.container.js]: Is this needed if nothing is added?

**Pull request merged**